src/Streamly/Coreutils/FileTest.hs

```diff
@@ -322,3 +361,67 @@ isOlderThan = compareModTime (<)
 -- Like coreutil @test file1 -ef file2@.
 isSameFile :: FilePath -> IO FileTest
 isSameFile = undefined
+
+data TimeStampMin = TimeStampMin Ordering Int
+
+data FileSize = FileSize Ordering Integer
+
+getLocalTime :: IO MilliSecond64
+getLocalTime = fromAbsTime <$> getTime Realtime
+
+compareFsTime :: (FileStatus -> EpochTime) -> TimeStampMin -> IO FileTest
```

Comment:
Something like  - `compareFileTime :: (FileStatus -> PosixTime) -> (FileStatus -> (POSIXTime -> POSIXTime -> Bool) -> FilePath -> IO FileTest` would make sense as a general API.

Then you could write: `compareModTime = compareFileTime Files.modificationTimeHiRes` and other functions also in the same way.